doc/cosma7_run.slurm

```diff
@@ -26,12 +30,18 @@ export OMP_NUM_THREADS=14
 export I_MPI_PIN=1
 export KMP_HW_SUBSET=28c,1t 
 
-nodes=16
+nodes=$(expr $SLURM_NTASKS / 2)
 # mpi must multiply to the number of mpi ranks
-mpi="4.2.2.2"
-rankpernode=2
-totrank=$(( nodes * rankpernode ))
+mpi="${SLURM_NTASKS}.1.1.1"
```

Comment:
@agb32 I think this changes the distribution of the problem so looks incorrect now. The distribution should be as even as possible with the highest numbers to the left. So for 64 MPI tasks (as in your example script) this should be:

```
4.4.2.2
```

Can you fix this and rerun the benchmarks?

Reply:
I assumed that just tuned the way the problem was split, rather than something that would change the output results (i.e. different fabric topologies would be optimised by different values for this).
If thats not the case, can you suggest the values that should be used for each case, e.g.:
1 node: 2.1.1.1
2 node: 2.2.1.1
4 node: 2.2.2.1
8 node: 2.2.2.2
16 node: 4.2.2.2
32 node: 4.4.2.2

Reply:
Those look right to me.

Reply:
I need to check with the developers if this makes a real difference or not but for consistency with the other runs, we should use common distributions in all cases. I will also add notes on this to the benchmark documentation.

Reply:
A quick check shows that it is definitely doing something different as the timers for the single node "2.1.1.1" and "1.1.1.2" show different patterns. Whether that translates to different performance is still an open question but I think it does mean we need to be consistent with distributions.

Reply:
If the setting doesn't affect the result of the code (which I think would be poor if it did - since ones communication method shouldn't change the scientific output of a code), then I think we'd want to select the value that is most appropriate for the service it is running on.

For example, it might be that 4.4.2.2 is best for hypercube, while 64.1.1.1 is best for fat tree or something like that - and by forcing benchmarks to use 4.4.2.2 on non-hypercube configurations, we'd be deliberately reducing performance...

So it would be good to get clarity from the developers before continuing.

Reply:
I think it is setting the size of the local domain for the weak scaling test in four dimensions. Remember, this is not "the scientific code" as, in this case, Grid is a library of data parallel routines that are used to create programs for QCD modelling, not a code per se. "The code" here is a benchmark written to call the routines in a way that allows you to understand the performance of the system in the regime that is interesting to researchers.

I will check with the developers but, in the meantime, if we want to update the results I think we should be cautious and be consistent with the distribution.